### ISSUE TITLE
should not allow tag-num if there is no tag

### ISSUE THREAD

**Issue by dwapstra** [mbarkhau/bumpver#151]
If a package version does not have a `tag`, `bumpver update --tag-num` should not be allowed as it could create incorrect versions.

Example:
```python
$ bumpver update --dry --tag-num
INFO    - Old Version: 21.1
INFO    - New Version: 21.11
--- bumpver.toml
+++ bumpver.toml
@@ -1,5 +1,5 @@
 [bumpver]
-current_version = "21.1"
+current_version = "21.11"
 version_pattern = "YY.MM[.PATCH][PYTAGNUM]"
 commit_message = "bump version {old_version} -> {new_version}"
 commit = true
--- src/mypkg/__init__.py
+++ src/mypkg/__init__.py
@@ -1,2 +1,2 @@
-__version__ = '21.1'
+__version__ = '21.11'
```

**Comment by mbarkhau** [comment]
That's a fair point. Are you interested in contributing a fix?

**Comment by dwapstra** [comment]
> That's a fair point. Are you interested in contributing a fix?

Sure, just need a pointer where to handle this.

**Comment by mbarkhau** [comment]
I think the most reasonable place to add this check would be in `bumpver.v2version.incr`: https://github.com/mbarkhau/bumpver/blob/master/src/bumpver/v2version.py#L724

Something similar to what is done at the end of the function:

```python
def incr(...):
    ...
    has_tag_part = cur_vinfo.tag != "final"
    if tag_num and not has_tag_part:
        logger.error("Invalid arguments, non-final --tag=<tag> is needed to use --tag-num.")
        return None
```

Tests can be added here: https://github.com/mbarkhau/bumpver/blob/master/test/test_version.py

**Comment by dwapstra** [comment]
Created PR https://github.com/mbarkhau/bumpver/pull/153

**Comment by mbarkhau** [comment]
Fixed with `pip install "bumpver==2021.1109"`.
